5_databases/week3/homework3

```diff
@@ -0,0 +1,140 @@
+-- 3.Get a review with any id, fx 1
+SELECT *
+FROM Review
+WHERE id = 1;
+-- 4.Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE Review
+SET title = 'update title', description = 'update description', meal_id = 1, stars = 5, created_date = CURDATE();
+-- 5.Delete a review with any id, fx 1
+DELETE FROM Review
+WHERE id = 1;
+-- Additional queries
+-- Now add a couple of different meals, reservations and reviews with different attributes. With those meals create the following queries
+INSERT INTO meal (title, description, location, when_date, max_reservations, price, created_date)
+VALUES ('FUFU', 'Delicious Starchy', 'Ghana', '2023-11-21', 10, 50.00, CURDATE()),
+('Fante Fante', 'Spicey fish stew with Kenkey(made of corn)', 'Cape Coast', '2023-12-31 15:41:20', 10, 40.00, CURDATE()),
+('Rød grød med fløde', 'Delicious Danish dessert', 'Copenhagen', '2023-11-22 18:00:00', 10, 80.00, CURDATE()),
+('Pasta Carbonara', 'Classic Italian pasta dish', 'Rome', '2023-11-23 19:30:00', 10, 60.00, CURDATE()),
+('Indonesian Meal', 'A delicious traditional Indonesian meal', 'Indonesia', '2023-12-12 13:10:00', 10, 90.00, CURDATE()),
+('Sushi Night', 'Fresh Japanese sushi', 'Tokyo', '2023-11-25 20:00:00', 12, 150.00, CURDATE()),
+('Burger', 'American burger', 'New York', '2023-11-28 19:11:10', 10, 150.00, CURDATE()),
+('Tacos', 'Mexican tacos', 'Mexico', '2023-12-01 10:00:00', 10, 110.00, CURDATE());
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES (5, 1, CURDATE(), '22334455', 'Sam Jones', 'samjones@gmail.com'),
+(6,2, CURDATE(), '32334455', 'Peter Tosh', petertosh@gmail.com '),
+(8,4, 2, CURDATE(), '9233455', 'Bob Marley', 'bobmarley@gmail.com'),
+(10,3, CURDATE(), '112334455', 'Sam james', '
+SELECT *
+FROM meal
+-- Functionality
+-- 1.Get meals that has a price smaller than a specific price fx 90
+SELECT *
+FROM meal
+WHERE price < 90;
+-- 2.Get meals that still has available reservations
+SELECT *
+FROM meal
+WHERE max_reservations > 0;
+-- 3.Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT *
+FROM meal
+WHERE title LIKE 'Rød grød med%';
+-- 4.Get meals that has been created between two dates
+SELECT *
+FROM meal
+WHERE created_date BETWEEN '2023-11-21' AND '2023-11-25';
+-- 5.Get only specific number of meals fx return only 5 meals
+SELECT *
+FROM meal
+LIMIT 5;
+-- 6.Get the meals that have good reviews
+SELECT *
+FROM meal
+INNER JOIN review ON meal.id = review.meal_id
+WHERE review.stars > 3;
+-- 7.Get reservations for a specific meal sorted by created_date
+SELECT *
+FROM reservation
+WHERE meal_id = 1
+ORDER BY created_date;
+-- 8.Sort all meals by average number of stars in the reviews
+SELECT meal.title, AVG(review.stars) AS average_stars
+FROM meal
+INNER JOIN review ON meal.id = review.meal_id
+GROUP BY meal.title
+ORDER BY average_stars DESC;
```

Comment:
The user tells you that not all meals appear in this output. It seems that only meals with a review are included. How can we fix it to include all meals?